### PR TITLE
FDP-252: Maven structure

### PR DESCRIPTION
**Pull request by sanderv**
…streams as profile

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -147,10 +78,47 @@
...
-    <module>cucumber-tests-platform-distributionautomation</module>
-    <module>cucumber-tests-platform-publiclighting</module>
-    <module>cucumber-tests-platform-smartmetering</module>
-    <module>cucumber-tests-platform-microgrids</module>
-
   </modules>
+
+  <profiles>
+    <profile>
+      <id>flexovl</id>
```
Comment (bvdzwet) [inline integration-tests/pom.xml:85]: The mechanism of separating builds into profiles looks good.
I am a little doubtful regarding the naming of the flexovl and smhe profiles.
The naming feels based on particular applications outside the scope of the platform.
In the context of GXF would it be better to choose names reflecting the domains instead (public lighting, smart metering, possibly abbreviated to something like pl and sm to ease typing the profile flags on the command line)?

File: osgp/platform/pom.xml [hunk 2]
```diff
@@ -140,4 +113,82 @@
     <module>osgp-secret-management</module>
```
Comment (smvdheijden) [inline osgp/platform/pom.xml:113]: Up till now secret management is only used for smart metering. I'm wondering if we should add this to the smart metering profile?

File: osgp/pom.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  <groupId>org.osgp</groupId>
+  <artifactId>osgp</artifactId>
+  <version>5.14.0-SNAPSHOT</version>
+  <packaging>pom</packaging>
+
+  <modules>
+    <module>platform</module>
+    <module>shared</module>
+    <module>protocol-adapter-iec60870</module>
```
Comment (smvdheijden) [inline osgp/pom.xml:17]: Consider moving this module to public lighting and distribution automation profile.

File: osgp/pom.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    <module>protocol-adapter-iec61850</module>
```
Comment (smvdheijden) [inline osgp/pom.xml:18]: Consider moving this module to public lighting, distribution automation, and microgrids profiles.

File: osgp/pom.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    <module>protocol-adapter-mqtt</module>
```
Comment (smvdheijden) [inline osgp/pom.xml:19]: Move this module to distribution automation profile.

File: osgp/shared/pom.xml [hunk 6]
```diff
@@ -94,15 +94,44 @@
...
     <module>osgp-dto</module>
-    <module>osgp-kafka-distributionautomation</module>
-    <module>osgp-kafka-logger</module>
     <module>osgp-ws-secret-management</module>
-    <module>osgp-ws-smartmetering</module>
     <module>osgp-ws-core</module>
     <module>osgp-ws-admin</module>
-    <module>osgp-ws-tariffswitching</module>
-    <module>osgp-ws-publiclighting</module>
     <module>osgp-ws-microgrids</module>
```
Comment (smvdheijden) [inline osgp/shared/pom.xml:100]: Move this module to microgrids profile.

File: osgp/shared/pom.xml [hunk 7]
```diff
@@ -94,15 +94,44 @@
     <module>parent-shared</module>
     <module>shared</module>
     <module>osgp-dto</module>
-    <module>osgp-kafka-distributionautomation</module>
-    <module>osgp-kafka-logger</module>
     <module>osgp-ws-secret-management</module>
```
Comment (smvdheijden) [inline osgp/shared/pom.xml:97]: Consider moving this module to smart metering profile.

File: pom.xml [hunk 8]
```diff
@@ -33,107 +33,8 @@
...
-    <module>integration-tests/cucumber-tests-platform</module>
-    <module>integration-tests/cucumber-tests-platform-common</module>
-    <module>integration-tests/cucumber-tests-platform-distributionautomation</module>
-    <module>integration-tests/cucumber-tests-platform-microgrids</module>
-    <module>integration-tests/cucumber-tests-platform-publiclighting</module>
-    <module>integration-tests/cucumber-tests-platform-smartmetering</module>
-    <module>integration-tests/cucumber-tests-protocol-iec60870</module>
-
-    <!-- demo client application -->
-    <module>public-lighting-demo-app/parent-os-webapps</module>
```
Comment (smvdheijden) [inline pom.xml:135]: Are the demo app modules completely removed?

File: pom.xml [hunk 9]
```diff
@@ -33,107 +33,8 @@
   </licenses>
 
   <modules>
-    <!-- super pom -->
-    <module>super/pom.xml</module>
```
Comment (smvdheijden) [inline pom.xml:37]: Is this module no longer needed here?

File: pom.xml [hunk 10]
```diff
@@ -33,107 +33,8 @@
...
-    <module>integration-tests/cucumber-tests-platform-distributionautomation</module>
-    <module>integration-tests/cucumber-tests-platform-microgrids</module>
-    <module>integration-tests/cucumber-tests-platform-publiclighting</module>
-    <module>integration-tests/cucumber-tests-platform-smartmetering</module>
-    <module>integration-tests/cucumber-tests-protocol-iec60870</module>
-
-    <!-- demo client application -->
-    <module>public-lighting-demo-app/parent-os-webapps</module>
-    <module>public-lighting-demo-app/web-demo-app</module>
+    <module>osgp</module>
```
Comment (smvdheijden) [inline pom.xml:36]: :+1:

**Review by smvdheijden — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 9 above)
Comment (sanderv) [reply pom.xml:37]: That module doesn't produce artifacts, it's referenced by other modules as super-pom. It shouldn't be built on its own.

File: integration-tests/pom.xml (hunk 1 above)
Comment (sanderv) [reply integration-tests/pom.xml:85]: As discussed: changed to 'pl'

File: pom.xml (hunk 8 above)
Comment (sanderv) [reply pom.xml:135]: Missed that one, put it back

**Review by smvdheijden — APPROVED** [review]

**Review by smvdheijden — APPROVED** [review]

**Pull request merged**